core/chain.go

```diff
@@ -22,5 +23,6 @@ type ChainConfig struct {
 	From         string            // address of key to use
 	KeystorePath string            // Location of key files
 	Insecure     bool              // Indicated whether the test keyring should be used
+	ChainLogger  log.Logger        // The logger for chain specific errors
```

Comment:
There is no need for the logger to be this high up in the call stack. Please initialize loggers inside of `IntializeChain` for ethereum and substrate and pass them directly into the Connection, Listener, and Writers.